cachi2/core/package_managers/gomod.py

```diff
@@ -317,7 +317,10 @@ def _resolve_gomod(app_dir: RootedPath, request: Request, tmp_dir: Path) -> dict
         # Make Go ignore the vendor dir even if there is one
         go_list.extend(["-mod", "readonly"])
 
-    main_module_name = _run_gomod_cmd([*go_list, "-m"], run_params).rstrip()
+
+    main_module_name = _get_module_name_from_repo_origin(app_dir)
+    if main_module_name is None:
+        main_module_name = _run_gomod_cmd([*go_list, "-m"], run_params).rstrip()
```

Comment:
Maybe we should report it like this?

```json
{
  "name": "name from go.mod",
  "purl": "pkg:golang/name from git origin"
}
```
There's also the question of what to do with package names. It would be pretty weird if the main module was `github.com/kubernetes/kubernetes` and all the package were `k8s.io/kubernetes/...`

Reply:
Does this look right?
```
{
    "name": "github.com/totally-not-real-path",
    "version": "v1.0.0",
    "purl": "pkg:golang/github.com/actual-module-path@v1.0.0"
},
{
    "name": "github.com/totally-not-real-path/package",
    "version": "v1.0.0",
    "purl": "pkg:golang/github.com/actual-module-path/package@v1.0.0"
},
{
    "name": "github.com/totally-not-real-replacement",
    "version": "v1.0.0",
    "purl": "pkg:golang/github.com/actual-module-path/actual-replacement-path@v1.0.0"
}
```

Considering the replacement is:
```
replace github.com/random-module => ./actual-replacement-path 
```

And the `go.mod` in `./actual-replacement-path`:
```
module github.com/totally-not-real-replacement

go 1.19
```

Reply:
I think Go wouldn't allow you to `replace github.com/random-module` with something that doesn't have `module github.com/random-module` in go.mod

But otherwise yes, that's what I would expect :+1: